### PR TITLE
Homework3: Krivosheev Konstantin

### PR DESCRIPTION
**Pull request by TexnoBY**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle)
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks
- [x] my PR made from the fork feature branch to maxim homework# branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: src/homework3/task1.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+"""
+Напишите программу, которая печатает цифры от 1 до 100,
+но вместо чисел, кратных 3 пишет Fizz,
+вместо чисел кратный 5 пишет Buzz,
+а вместо чисел одновременно кратных и 3 и 5 - FizzBuzz
+"""
+
+
+def fizz_buzz():
+    [print("Fizz" * (not i % 3) + "Buzz" * (not i % 5) or i)
```
Comment (belovmd) [inline src/homework3/task1.py]:
```
[
    text
]
```
is a usual code style for this

File: src/homework3/task2.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+4. Одной строкой удалите элемент  '2a' из прошлого списка и напечатайте его.
+
+5. Скопируйте список и добавьте в него элемент '2a' так чтобы в исходном
+ списке этого элемента не было.
+"""
+
+
+def list_practice():
+    # 1
+    lst1 = [chr1 + chr2 for chr1 in ['a', 'b'] for chr2 in ['b', 'c', 'd']]
```
Comment (belovmd) [inline src/homework3/task2.py]: 'ab'?

File: src/homework3/task2.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    print(lst1)
+
+    # 2
+    lst2 = lst1[::2]
+    print(lst2)
+
+    # 3
+    lst3 = [str(chr1) + 'a' for chr1 in range(1, 5)]
```
Comment (belovmd) [inline src/homework3/task2.py]: '1234'?

File: src/homework3/task3.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def tuple_practice():
+    # 1
+    lst1 = ['a', 'b', 'c']
+    tuple1 = tuple(lst1)
+    print(tuple1)
+
+    # 2
+    tuple2 = 'a', 'b', 'c'
```
Comment (belovmd) [inline src/homework3/task3.py]: tuple = (a, b, c) is a usual style. But it works in your way too

File: src/homework3/task4.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+Важно: 1 1 1 - это 3 пары, 1 1 1 1 - это 6 пар
+"""
+
+
+def count_pairs_simple(lst):
+    count = 0
+    for start in range(len(lst)):
+        for el in lst[start + 1:]:
+            if lst[start] == el:
+                count += 1
```
Comment (belovmd) [inline src/homework3/task4.py]: O(n**2) solution

File: src/homework3/task4.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    return count
+
+
+def count_pairs_advanced(lst):
+    count_elements = {}
+    for el in lst:
+        count_elements[el] = count_elements.get(el, 0) + 1
+    return sum([sum(range(value)) for value in count_elements.values()])
```
Comment (belovmd) [inline src/homework3/task4.py]: O(n**2)

File: src/homework3/task4.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+
+if __name__ == '__main__':
```
Comment (belovmd) [inline src/homework3/task4.py:18]: has O(1) solution

File: src/homework3/task6.py [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+ часть списка, не меняя их порядок, а все нули - в правую часть.
+Порядок ненулевых элементов изменять нельзя, дополнительный список
+ использовать нельзя, задачу нужно выполнить за один проход по списку.
+Распечатайте полученный список.
+"""
+
+
+def zero_to_right(lst):
+    count = 0
+    for i in range(len(lst)):
```
Comment (belovmd) [inline src/homework3/task6.py:12]: range(len(lst)) it's bad practice

Comment (TexnoBY) [reply src/homework3/task6.py:12]: In my solution it is necessary

File: src/homework3/task4.py [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+Выходные данные - количество пар.
+Важно: 1 1 1 - это 3 пары, 1 1 1 1 - это 6 пар
+"""
+
+
+def count_pairs(lst):
+    count_elements = {}
+    for el in lst:
+        count_elements[el] = count_elements.get(el, 0) + 1
+    return sum(value * (value - 1) // 2 for value in count_elements.values())
```
Comment (belovmd) [inline src/homework3/task4.py:15]: great

File: src/homework3/task6.py (hunk 8 above)
Comment (belovmd) [reply src/homework3/task6.py:12]: yes, but it has simplier solution. 

``` python
zero_counter = 0
for element in last
    if not element:
          linst.remove
# and next - list.extend([0]*zero_count)
```

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**